cl/users/email_handlers.py

```diff
@@ -676,3 +677,83 @@ def enqueue_email(recipients: list[str], message_id: str) -> None:
+    "current email address. "
+    "Please consider changing your email address.",
+    EMAIL_NOTIFICATIONS.NO_EMAIL: "Your email provider sent a bounce "
+    "message indicating that your email address doesn't exist. Please "
+    "consider changing your email address.",
+    EMAIL_NOTIFICATIONS.COMPLAINT: "We've received a complaint from "
+    "your email provider indicating that you don't want to receive emails "
+    "from us. Please consider changing your email address.",
+}
+
+TRANSIENT = {
+    EMAIL_NOTIFICATIONS.GENERAL: "Your email provider sent a bounce message. "
+    "We are still attempting to fix this problem by resending "
+    "messages.",
+    EMAIL_NOTIFICATIONS.MAILBOX_FULL: "Your email provider sent a "
+    "bounce message because your inbox was full. Please consider clearing out "
+    "your email inbox.",
+}
+
+
+def determine_email_error_message(type: str, subtype: int) -> str:
+    """Determine the message to display to the user with more details
+    about the Permanent or Transient email address error.
+
+    :param type: The type of email address error, Permanent or Transient.
+    :param subtype: The error subtype returned by the SES notification.
+    :return str: The error message with more details.
+    """
+
+    if type == "PERMANENT":
+        return PERMANENT.get(subtype, PERMANENT[EMAIL_NOTIFICATIONS.GENERAL])
+    else:
+        return TRANSIENT.get(subtype, TRANSIENT[EMAIL_NOTIFICATIONS.GENERAL])
+
+
+def broken_email_address(user: User | AnonymousUser) -> dict:
+    """Determine if the user's email address has a conflict and the type of
+    error, Permanent or Transient.
+
+    :param request: The HttpRequest.
+    :return dict: A dict with a tuple that returns the following values:
+    0: (email broken) True or False
+    1: (Type of error) Permanent or Transient
+    2: The user's email address with the error
+    3: Datetime when the error started.
+    4: The error sub-type to provide more details to the user.
+    """
+
+    if user.is_authenticated:
+        email_banned = EmailFlag.objects.filter(
+            email_address=user.email, object_type=OBJECT_TYPES.BAN
+        )
+        backoff_event = BackoffEvent.objects.filter(email_address=user.email)
+
+        if email_banned.exists():
+            return {
+                "EMAIL_BROKEN_BANNER": {
+                    "show": True,
+                    "type": "PERMANENT",
+                    "date": email_banned[0].date_created,
+                    "sub_type": email_banned[0].event_sub_type,
+                }
+            }
+
+        if backoff_event.exists():
+            return {
+                "EMAIL_BROKEN_BANNER": {
+                    "show": True,
+                    "type": "TRANSIENT",
+                    "date": backoff_event[0].date_created,
+                    "sub_type": backoff_event[0].event_sub_type,
+                }
+            }
+
+    return {"EMAIL_BROKEN_BANNER": {"show": False}}
```

Comment:
I think that most of this function can be removed.

Reply:
Or more specifically I dont see the need for the giant dictionary.  Instead we could do something like this 

```    
    msg = None
    email_banned = EmailFlag.objects.filter(
        email_address=user_email, object_type=OBJECT_TYPES.BAN
    )
    if email_banned.exists():
        return PERMANENT.get(
            email_banned[0].event_sub_type,
            PERMANENT[EMAIL_NOTIFICATIONS.GENERAL],
        )
        
    backoff_event = BackoffEvent.objects.filter(email_address=user_email)
    if backoff_event.exists():
        return TRANSIENT.get(
            backoff_event[0].event_sub_type,
            TRANSIENT[EMAIL_NOTIFICATIONS.GENERAL],
        )
    return msg
```

We only need to return the message I think.